src/v/cluster/controller_backend.cc

```diff
@@ -916,6 +915,30 @@ controller_backend::process_partition_reconfiguration(
     co_return ec;
 }
 
+bool controller_backend::can_finish_update(
+  topic_table_delta::op_type update_type,
+  const std::vector<model::broker_shard>& current_replicas,
+  const std::vector<model::broker_shard>& previous_replicas) {
+    // force abort update may be finished by any node
+    if (update_type == topic_table_delta::op_type::force_abort_update) {
+        return true;
+    }
+    // update may be finished by a node that was added to replica set
+    if (!has_local_replicas(_self, previous_replicas)) {
```

Comment:
nit: just looking at this method, this comment doesn't quite match up with the behavior. Could you add a note that it's expected that one of `current_replicas` is local?